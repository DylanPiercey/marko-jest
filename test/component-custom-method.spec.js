const path = require('path');
const { initComponent, createTestSandbox } = require('../test-utils');

describe('component with custom methods', () => {
  const componentClass = initComponent(path.resolve(__dirname, './resources/fake-video/index.marko'));

  let testSandbox;
  let component;

  beforeEach(async () => {
    testSandbox = createTestSandbox();
    component = await testSandbox.renderComponent(componentClass, {});
  });

  afterEach(() => {
    testSandbox.reset();
  });

  it('should render properly', () => {
    expect(component.els).toMatchSnapshot();
  });

  it('should able to call its method from the component instance', async () => {
    component.playVideo();
    component.update();
    expect(component.els).toMatchSnapshot();

    component.stopVideo();
    component.update();
    expect(component.els).toMatchSnapshot();
  });
});
