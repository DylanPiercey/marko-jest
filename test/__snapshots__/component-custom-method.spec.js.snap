// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component with custom methods should able to call its method from the component instance 1`] = `
Array [
  <p>
    Enjoy the movie 
    <span>
      üòéüçø
    </span>
  </p>,
]
`;

exports[`component with custom methods should able to call its method from the component instance 2`] = `
Array [
  <p />,
]
`;

exports[`component with custom methods should render properly 1`] = `
Array [
  <p />,
]
`;
